### PR TITLE
display coding activity into status bar

### PR DESCRIPTION
**Pull request by gandarez**
* display coding activity into status bar
* created new command `wakatime.status_bar_coding_activity`
* caching api key to reduce I/O cost
* added Travis integration
* small fixes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,23 +36,23 @@
         "url": "https://github.com/wakatime/vscode-wakatime.git"
     },
     "engines": {
-        "vscode": "^1.17.0",
+        "vscode": "^1.25.0",
```
Comment (alanhamlett) [inline package.json]: Shouldn't this stay at `^1.17.0`?

File: package.json [hunk 2]
```diff
@@ -36,23 +36,23 @@
...
         "postinstall": "node ./node_modules/vscode/bin/install",
-        "test": "npm run compile-dev && node ./node_modules/vscode/bin/test"
+        "test": "npm run compile:dev && node ./node_modules/vscode/bin/test"
     },
     "contributes": {
         "commands": [
             {
                 "command": "wakatime.apikey",
-                "title": "WakaTime: Api Key"
+                "title": "WakaTime: Set Api Key"
```
Comment (alanhamlett) [inline package.json]: Let's leave it as `Api Key` because it can also be used to view the key not just set it.

File: package.json [hunk 3]
```diff
@@ -64,7 +64,11 @@
             },
             {
                 "command": "wakatime.status_bar_icon",
-                "title": "WakaTime: Status Bar Icon"
+                "title": "WakaTime: Set Status Bar Icon"
```
Comment (alanhamlett) [inline package.json]: Let's remove `Set`.

File: package.json [hunk 4]
```diff
@@ -64,7 +64,11 @@
...
+            },
+            {
+                "command": "wakatime.status_bar_coding_activity",
+                "title": "WakaTime: Set Status Bar Coding Activity"
```
Comment (alanhamlett) [inline package.json]: Let's remove `Set`.

File: package.json [hunk 5]
```diff
@@ -87,16 +91,19 @@
...
-        "request": "2.88.0",
-        "rimraf": "^2.6.3",
         "ts-loader": "^5.3.3",
         "tslib": "^1.9.3",
         "typescript": "^3.4.3",
         "vscode": "^1.1.33",
         "webpack": "^4.30.0",
         "webpack-cli": "^3.3.0"
+    },
+    "dependencies": {
```
Comment (alanhamlett) [inline package.json]: These need to be moved back into `devDependencies` or else VS Code will download and install these packages when installing the wakatime plugin, which isn't needed and can cause network errors when installing the plugin.

File: package.json (hunk 1 above)
Comment (gandarez) [reply package.json]: No, because `ViewColumn.Beside` was introduced in the version 1.25.0

**Comment by gandarez** [comment]
@alanhamlett what about `Set Debug` and `Set Proxy`?

File: src/stats.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        this.options.getSetting('settings', 'api_key', (_, val) => { this.apiKey = val; });
+        this.options.getSetting('settings', 'proxy', (_, val) => { this.proxy = val; });
+    }
+
+    public async getCodingActivity(): Promise<string> {
+        return new Promise<string>(async (resolve, reject) => {
+            if (Libs.validateKey(this.apiKey)) return reject('Invalid Api Key');
+            const request = await import('request');
+            let date = new Date();
+            let dateStr = `${date.getFullYear()}-${date.getMonth() + 1}-${date.getDate()}`;
```
Comment (alanhamlett) [inline src/stats.ts]: The date format must add zeros when month/day less than 9, for ex: `2019-01-01`

File: package.json (hunk 1 above)
Comment (alanhamlett) [reply package.json]: Ok cool.

**Comment by alanhamlett** [comment]
> what about Set Debug and Set Proxy should be also removed?

Yes, let's remove `Set` there too.

File: package.json (hunk 5 above)
Comment (gandarez) [reply package.json]: fixed! Thanks for letting me know this "issue" regarding dev and non-dev dependencies into npm.

File: src/stats.ts (hunk 6 above)
Comment (gandarez) [reply src/stats.ts]: done!

File: src/stats.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        this.options.getSetting('settings', 'api_key', (_, val) => { this.apiKey = val; });
+        this.options.getSetting('settings', 'proxy', (_, val) => { this.proxy = val; });
+    }
+
+    public async getCodingActivity(): Promise<string> {
+        return new Promise<string>(async (resolve, reject) => {
+            if (Libs.validateKey(this.apiKey)) return reject('Invalid Api Key');
+            const request = await import('request');
+            let date = new Date();
+            let dateStr = `${date.getFullYear()}-${Libs.zeroFill(date.getMonth() + 1, 2)}-${Libs.zeroFill(date.getDate(), 2)}`;
```
Comment (alanhamlett) [inline src/stats.ts]: Let's create a new helper function named `getToday` like this:

```typescript
public static getToday(): string {
  var date = new Date();
  return date.toISOString().slice(0,10);
}
```

**Pull request merged**